cookbook/routing/extending_routing.rst

```diff
@@ -0,0 +1,15 @@
+Extending the Routing System
```

Comment:
I like the idea, but I think adding articles to the cookbook and putting "how do I...?" entires in the cookbook TOC is much more useful - e.g. potentially "How do I convert controller arguments to objects" - which would be something very short pointing to FrameworkExtraBundle. That's arguably overkill... but more findable.